go_dependencies.bzl

```diff
@@ -37,219 +36,213 @@ def bb_storage_go_dependencies():
-        strip_prefix = "go-grpc-prometheus-1.2.0",
-        urls = ["https://github.com/grpc-ecosystem/go-grpc-prometheus/archive/v1.2.0.tar.gz"],
+        sum = "h1:Ovs26xHkKqVztRpIrF/92BcuyuQ/YW4NSIpoGtfXNho=",
+        version = "v1.2.0",
     )
 
     go_repository(
         name = "com_github_lazybeaver_xorshift",
-        commit = "ce511d4823dd074d7c37a74225320332d6961abb",
         importpath = "github.com/lazybeaver/xorshift",
+        sum = "h1:TfmftEfB1zJiDTFi3Qw1xlbEbfJPKUhEDC19clfBMb8=",
+        version = "v0.0.0-20170702203709-ce511d4823dd",
     )
 
     go_repository(
         name = "com_github_matttproud_golang_protobuf_extensions",
-        commit = "c12348ce28de40eed0136aa2b644d0ee0650e56c",
         importpath = "github.com/matttproud/golang_protobuf_extensions",
+        sum = "h1:4hp9jkHxhMHkqkrB3Ix0jegS5sx/RkqARlsWZ6pIwiU=",
+        version = "v1.0.1",
     )
 
     go_repository(
         name = "com_github_prometheus_client_golang",
         importpath = "github.com/prometheus/client_golang",
-        sha256 = "e255f632b7223f794e0f60d99976535c16153cc00fadf39ee87fd9f678d6a32c",
-        strip_prefix = "client_golang-1.6.0",
-        urls = ["https://github.com/prometheus/client_golang/archive/v1.6.0.tar.gz"],
+        sum = "h1:NTGy1Ja9pByO+xAeH/qiWnLrKtr3hJPNjaVUwnjpdpA=",
+        version = "v1.7.1",
     )
 
     go_repository(
         name = "com_github_cespare_xxhash_v2",
         importpath = "github.com/cespare/xxhash/v2",
-        sha256 = "0ee31178d2c5a1249be4e26294a2f428008dc4e1ecbbfbe47f74e41026df1148",
-        strip_prefix = "xxhash-2.1.1",
-        urls = ["https://github.com/cespare/xxhash/archive/v2.1.1.tar.gz"],
+        sum = "h1:6MnRN8NT7+YBpUIWxHtefFZOKTAPgGjpQSxqLNn0+qY=",
+        version = "v2.1.1",
     )
 
     go_repository(
         name = "com_github_prometheus_client_model",
         importpath = "github.com/prometheus/client_model",
-        sha256 = "4ab1be9cdfa702d7f49beeb09a256bcc6a2aad55e8a0a37e7732a46934264e12",
-        strip_prefix = "client_model-0.2.0",
-        urls = ["https://github.com/prometheus/client_model/archive/v0.2.0.tar.gz"],
+        sum = "h1:uq5h0d+GuxiXLJLNABMgp2qUWDPiLvgCzz2dUR+/W/M=",
+        version = "v0.2.0",
     )
 
     go_repository(
         name = "com_github_prometheus_common",
-        commit = "6225330c4d5d7d0ffa21a82ad7eb041f82f25c7f",
         importpath = "github.com/prometheus/common",
+        sum = "h1:RyRA7RzGXQZiW+tGMr7sxa85G1z0yOpM1qq5c8lNawc=",
+        version = "v0.10.0",
     )
 
     go_repository(
         name = "com_github_prometheus_procfs",
         importpath = "github.com/prometheus/procfs",
-        sha256 = "7f31adcafbbfd7cb2e5d5b5954a3f55565ad41fb039c977b1f769c7d3dfc2a60",
-        strip_prefix = "procfs-0.0.10",
-        urls = ["https://github.com/prometheus/procfs/archive/v0.0.10.tar.gz"],
+        sum = "h1:F0+tqvhOksq22sc6iCHF5WGlWjdwj92p0udFh1VFBS8=",
+        version = "v0.1.3",
```

Comment:
That's correct, but turns out this is a net improvement due to using version tags instead of commit hashes (due to git protocol limitations, you can't shallow clone a commit hash)

Reply:
As discussed on Slack, with the `sum =`, it can do an efficient shallow clone. It's even more bandwidth efficient now.